### PR TITLE
python binding for velocity based object control

### PR DESCRIPTION
**Pull request by pushkalkatara**
## Motivation and Context

Extending Python bindings to control object's linear and angular velocities.

https://github.com/facebookresearch/habitat-api/issues/356

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by erikwijmans** [comment]
Thank you for the PR.  Please see https://github.com/facebookresearch/habitat-sim#code-style for fixing the cpp_lint test.

**Comment by pushkalkatara** [comment]
@erikwijmans Thanks for pointing out. Fixed the cpp_lint.

**Comment by aclegg3** [comment]
> you'll want to look at the [VelocityControl ](https://github.com/facebookresearch/habitat-sim/blob/1950ea2067ecab797ec65893d95ad75b3daf1a99/src/esp/physics/RigidObject.h#L90)structure. 

See PR #571 
I've added bindings for this now, so once this is ready all velocity functionality should be available in python. 😄 Thanks for prompting this!

**Comment by pushkalkatara** [comment]
Thanks a lot for the prompt response @aclegg3 

I was confused between bullet based `MotionType.DYNAMIC` and `MotionType.KINETIC`.

There are getter/setter functions for bullet based control defined here as well- https://github.com/facebookresearch/habitat-sim/blob/ae6ba1cdc772f7a5dedd31cbf9a5b77f6de3ff0f/src/esp/physics/bullet/BulletRigidObject.cpp#L396 . 

Is there a factor of accuracy while applying velocities with `MotionType.DYNAMIC` and `MotionType.KINETIC`. While using with the agent, would it be more accurate if used with `MotionType.DYNAMIC` enabled?

File: tests/test_physics.py [hunk 1]
```diff
@@ -71,6 +72,35 @@ def test_kinematics(sim):
     assert np.allclose(sim.get_transformation(object_id), expected)
     assert np.allclose(quat_from_magnum(sim.get_rotation(object_id)), Q)
 
+    previous_object_states = [
+        [sim.get_translation(object_id), sim.get_rotation(object_id)],
+        [sim.get_translation(object2_id), sim.get_rotation(object2_id)],
+    ]
+
+    # test get and set linear velocity
+    for _ in range(50):
```
[the remark this answers is not in the document]
Comment (pushkalkatara) [reply tests/test_physics.py:81]: Thanks for pointing out.

https://github.com/facebookresearch/habitat-sim/pull/571

**Comment by aclegg3** [comment]
> Is there a factor of accuracy while applying velocities with `MotionType.DYNAMIC` and `MotionType.KINETIC`. 

Yes, there are a few key differences.

1. You can use `MotionType.KINEMATIC` without Bullet installed via the VelocityControl structure. There will be no collision detection (the agent can move through walls, floors, etc...) if commanded and will not be constrained to move on the NavMesh without some modifications. The setter functions wrapped in this PR will not affect `MotionType.KINEMATIC` objects.

2. An object with `MotionType.DYNAMIC` will still be affected by collisions, gravity, damping, etc... and you will only be setting the **initial** velocity before each simulation step. Don't expect accurate velocity control via this method unless you account for this by stepping the simulation more often for smaller timesteps. e.g. ( `for(time): sim.step_world(0.00416)` )

**Comment by pushkalkatara** [comment]
Solved by https://github.com/facebookresearch/habitat-sim/pull/571

**Comment by pushkalkatara** [comment]
Right okay, I think best would be to use `MotionType.KINEMATIC` without bullet compiled and control the agent via VelocityControl structure.

**Pull request closed without merging**